src/Http/Request/Body/ImagesPropertyPolyfillRequestBodyParser.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Request\Body;
+
+use Broadway\Repository\AggregateNotFoundException;
+use CultuurNet\UDB3\Http\ApiProblem\ApiProblem;
+use CultuurNet\UDB3\Http\ApiProblem\SchemaError;
+use CultuurNet\UDB3\Iri\IriGeneratorInterface;
+use CultuurNet\UDB3\Media\MediaObject;
+use CultuurNet\UDB3\Media\MediaObjectRepository;
+use CultuurNet\UDB3\Model\ValueObject\MediaObject\MediaObjectIDParser;
+use CultuurNet\UDB3\Model\ValueObject\Web\Url;
+use InvalidArgumentException;
+use Psr\Http\Message\ServerRequestInterface;
+use stdClass;
+
+/**
+ * Polyfills the mediaObject/images properties on incoming JSON-LD of events, places and organizers that need to be
+ * imported if they only contain an @id. Also generates an @id if missing but id is given.
+ */
+final class ImagesPropertyPolyfillRequestBodyParser implements RequestBodyParser
+{
+    private const EVENTS = 'mediaObject';
+    private const PLACES = 'mediaObject';
+    private const ORGANIZERS = 'images';
+
+    private string $propertyName;
+    private IriGeneratorInterface $iriGenerator;
+    private MediaObjectRepository $mediaObjectRepository;
+
+    private function __construct(
+        string $propertyName,
+        IriGeneratorInterface $imagesIriGenerator,
+        MediaObjectRepository $mediaObjectRepository
+    ) {
+        $this->propertyName = $propertyName;
```

Comment:
Maybe change this to `imagePropertyName`?

Reply:
Done!